lib/data/external/AzureClient.js

```diff
@@ -39,14 +39,34 @@ class AzureClient {
           azureKey, stream, size, options, err => {
               if (err) {
                   const log = createLogger(reqUids);
-                  log.error('err from Azure data backend',
+                  log.error('err from Azure PUT data backend',
                   { error: err.message, stack: err.stack,
                     dataStoreName: this._dataStoreName });
                   return callback(errors.InternalError);
               }
               return callback(null, azureKey);
           });
     }
+
+    get(objectGetInfo, range, reqUids, callback) {
+        // for backwards compatibility
+        const key = typeof(objectGetInfo) === 'string' ? objectGetInfo :
+          objectGetInfo.key;
+        const response = objectGetInfo.response;
+        const azureStreamingOptions = objectGetInfo.azureStreamingOptions;
+        this._client.getBlobToStream(this._azureContainerName, key, response,
+          azureStreamingOptions, err => {
+              if (err) {
+                  const log = createLogger(reqUids);
+                  log.error('err from Azure GET data backend',
```

Comment:
does this actually log an error "not found" on your error test? Even if it does, we should consider doing a head call to azure first to see if we are going to get back an error so we can fail more gracefully.